packages/components/src/internal/components/settings/DeleteProjectModal.tsx

```diff
@@ -0,0 +1,208 @@
+                    assay data, etc.) in other projects.
+                </p>
+            </div>
+
+            <b> Project Data </b>
+
+            {summaries ? (
+                <table className="table table-responsive table-condensed delete-project-modal__table ">
+                    <tbody>
+                        {summaries.map(s => (
+                            <tr key={s.count + s.noun}>
+                                <td>{s.noun}</td>
+                                <td>{s.count}</td>
+                            </tr>
+                        ))}
+                    </tbody>
+                </table>
+            ) : (
+                <LoadingSpinner wrapperClassName="delete-project-modal__spinner" />
+            )}
+
+            <CommentArea comment={comment} onChangeComment={onChangeComment} />
+        </Modal.Body>
+    );
+});
+
+interface BodyDeletingProps {
+    totalCountFromSummaries: number;
+}
+export const BodyDeleting: FC<BodyDeletingProps> = memo(({ totalCountFromSummaries }) => {
+    const [toggle, seToggle] = React.useState<boolean>(false);
+
+    React.useEffect(() => {
+        setTimeout(() => seToggle(true), 1);
+    }, []);
+
+    return (
+        <Modal.Body>
+            <div className="deleting-project-modal-text">Please don't close this page until deleting is done.</div>
+            <Progress delay={0} estimate={totalCountFromSummaries * 15} toggle={toggle} />
+        </Modal.Body>
+    );
+});
+
+interface Props {
+    onCancel: () => void;
+    onError: (e: string) => void;
+    projectName: string;
+}
+
+export const DeleteProjectModal: FC<Props> = memo(props => {
+    const { projectName, onCancel, onError } = props;
+    const [summaries, setSummaries] = useState<Summary[]>(undefined);
+    const [comment, setComment] = useState<string>(undefined);
+    const [isDeleting, setIsDeleting] = useState<boolean>(false);
+
+    const { api } = useAppContext();
+    const { container, user } = useServerContext();
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const allSummaries = await api.security.getDeletionSummaries();
+                setSummaries(allSummaries);
+            } catch (e) {
+                // getDeletionSummaries() handles error logging
+                // Purposely leave loading spinner indefinitely in case of summary retrieval failure
+            }
+        })();
+    }, [api.security]);
+
+    const onHide = useCallback(() => {
+        if (!isDeleting) onCancel();
+    }, [isDeleting, onCancel]);
+
+    const onChangeComment = useCallback(e => {
+        setComment(e.target.value);
+    }, []);
+
+    const onDeleteProject = useCallback(async () => {
+        setIsDeleting(true);
+        try {
+            await api.security.deleteContainer(comment);
+
+            const successMsg = `${projectName} successfully deleted.`;
+            const adminProjectsHref = createProductUrl(
```

Comment:
Surely there is a better way to construct these two urls that I missed while trawling helper functions.